dangerzone/isolation_provider/base.py

```diff
@@ -102,6 +131,113 @@ def sanitize_conversion_str(self, untrusted_conversion_str: str) -> str:
         return armor_start + conversion_string + armor_end
 
 
+class ProcessBasedIsolationProvider(IsolationProvider):
```

Comment:
I'd suggest merging `ProcessBasedIsolationProvider` into `IsolationProvider`, for the following reasons:

1. We don't have another type of isolation provider, so we don't benefit from the extra level of abstraction, and we hinder code-readability.
2. We can make the `Dummy` provider run closer to the way the real isolation providers will run.

Reply:
We decided not to go with the UnsafeConverter yet, until it can run in all platforms.